ambry-api/src/main/java/com/github/ambry/config/RouterConfig.java

```diff
@@ -656,6 +656,24 @@ public class RouterConfig {
   @Default("false")
   public final boolean routerRepairWithReplicateBlobEnabled;
 
+  /**
+   * The maximum duration in seconds to retry. If the get blob operation takes more than this duration, we would not retry.
+   */
+  @Config(ROUTER_GET_BLOB_RETRY_LIMIT_IN_SEC)
+  @Default("0")
+  public final int routerGetBlobRetryLimitInSec;
+  public static final String ROUTER_GET_BLOB_RETRY_LIMIT_IN_SEC = "router.get.blob.retry.limit.in.sec";
+  public static final int ROUTER_GET_BLOB_RETRY_LIMIT_IN_SEC_MAX = 300;
+
+  /**
+   * The maximum number of retries for get blob. The default value is 0, which means no retry for get blob.
+   */
+  @Config(ROUTER_GET_BLOB_RETRY_LIMIT_COUNT)
+  @Default("0")
+  public final int routerGetBlobRetryLimitCount;
+  public static final String ROUTER_GET_BLOB_RETRY_LIMIT_COUNT = "router.get.blob.retry.limit.count";
+  public static final int ROUTER_GET_BLOB_RETRY_LIMIT_COUNT_MAX = 100;
```

Comment:
100 seems high

Reply:
+1 Maybe 3 or 5 is good enough?

Reply:
this is the maximum number, I don't see we are setting the value to this number. But I don't want to limit it under 100.